tf2onnx/onnx_opset/tensor.py

```diff
@@ -2043,6 +2043,337 @@ def version_11(cls, ctx, node, **kwargs):
+            min_col = min_col_graph.make_node("Add", [diag_width.output[0], sup_min.output[0]])
+            min_col_graph.add_graph_output(min_col.output[0], ctx.get_dtype(col), [-1])
+            return min_col_graph
+
+        def MinRowGraph():
+            raw_sub_max = ctx.make_node("Min", [const_zero.output[0], k_max.output[0]])
+            sub_max = ctx.make_node("Abs", [raw_sub_max.output[0]])
+            min_row_graph = ctx.create_new_graph_with_same_config()
+            min_row_graph.parent_graph = ctx
+            min_row = min_row_graph.make_node("Add", [diag_width.output[0], sub_max.output[0]])
+            min_row_graph.add_graph_output(min_row.output[0], ctx.get_dtype(row), [-1])
+            return min_row_graph
+
+        col_not_set = ctx.make_node("Equal", [col, const_neg_one.output[0]])
+        col_set = ctx.make_node("Not", [col_not_set.output[0]])
+        row_not_set = ctx.make_node("Equal", [row, const_neg_one.output[0]])
+        row_set = ctx.make_node("Not", [row_not_set.output[0]])
+        need_min_col = ctx.make_node("And", [col_not_set.output[0], row_set.output[0]])
+        need_min_row = ctx.make_node("And", [row_not_set.output[0], col_set.output[0]])
+        final_col = ctx.make_node("If", [need_min_col.output[0]])
+        final_col.set_body_graph_as_attr("then_branch", MinColGraph())
+        final_col.set_body_graph_as_attr("else_branch", MaxColGraph())
+        final_row = ctx.make_node("If", [need_min_row.output[0]])
+        final_row.set_body_graph_as_attr("then_branch", MinRowGraph())
+        final_row.set_body_graph_as_attr("else_branch", MaxRowGraph())
+
+        '''next, compute lengths of all possilbe k into k_lens, e.g. if target matrix is of shape [2,2],
+        k_lens will be [1,2,1]'''
+        diag_max = ctx.make_node("Sub", [final_col.output[0], const_one.output[0]])
+        diag_min = ctx.make_node("Sub", [const_one.output[0], final_row.output[0]])
+        raw_row_col_diff = ctx.make_node("Sub", [final_col.output[0], final_row.output[0]])
+        row_col_diff = ctx.make_node("Abs", [raw_row_col_diff.output[0]])
+        row_col_diff_inc = ctx.make_node("Add", [row_col_diff.output[0], const_one.output[0]])
+        final_diag_len = ctx.make_node("Min", [final_col.output[0], final_row.output[0]])
+        final_diag_len_dec = ctx.make_node("Sub", [final_diag_len.output[0], const_one.output[0]])
+        # k_iter_1 is length of diagonals in right top corner
+        k_iter_1 = ctx.make_node("Range", [const_one.output[0], final_diag_len.output[0]], domain="com.microsoft")
+        # k_iter_2 is length of diagonals in the middle
+        k_iter_2 = ctx.make_node("Expand", [final_diag_len.output[0], row_col_diff_inc.output[0]])
+        # k_iter_3 is length of diagonals in left bottom corner
+        k_iter_3 = ctx.make_node("Range", [final_diag_len_dec.output[0], const_zero.output[0], const_neg_one.output[0]],
+                                 domain="com.microsoft")
```

Comment:
Should not use domain="com.microsoft" anymore -- since Range is an official function now.

Reply:
It is my intention to go with contrib one - since the standard ask for tensor of shape [] which requires few more lines to squeeze all inputs, the code will be further bloated.
Further, the standard one give no default value on unit of increment ... not the best design I should say.

Reply:
I think it's a hard requirement  (e.g. "com.microsoft"domain is not supported by platforms like TensorRT).

Reply:
Talking about standard, we shouldn't have MatrixDiagV3 in the first place.